### PR TITLE
Profile sync audit changes

### PR DESCRIPTION
**Pull request by KOTLAPRA23**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-1640

### Change description ###

feat: Profile sync audit changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/profilesync/schedular/UserProfileSyncJobScheduler.java [hunk 1]
```diff
@@ -41,55 +41,59 @@
...
+            log.error("{}::Sync Batch Job Failed::{}", loggingComponentName, e.getErrorMessage());
+            syncAudit.setSchedulerStatus("fail");
+            syncAudit.setSchedulerStartTime(startTime);
+            profileSyncAuditRepository.save(syncAudit);
 
         }
     }
 
-    private String getLastBatchFailureTimeInHours(LocalDateTime lastSuccessBatch) {
+    public String getLastSuccessfulRunInHours(LocalDateTime lastSuccessBatch) {
```
Comment (shashank-rastogi) [inline src/main/java/uk/gov/hmcts/reform/profilesync/schedular/UserProfileSyncJobScheduler.java:97]: rename to getLastSuccessTimeInHours()

Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/profilesync/schedular/UserProfileSyncJobScheduler.java:97]: ok.

File: src/test/java/uk/gov/hmcts/reform/profilesync/util/JsonFeignResponseUtilTest.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.util.MultiValueMap;
 import uk.gov.hmcts.reform.profilesync.client.IdamClient;
-import uk.gov.hmcts.reform.profilesync.domain.response.UserProfileResponse;
+import uk.gov.hmcts.reform.profilesync.domain.response.GetUserProfileResponse;
 import uk.gov.hmcts.reform.profilesync.helper.MockDataProvider;
```
[the remark this answers is not in the document]
Comment (KOTLAPRA23) [reply src/test/java/uk/gov/hmcts/reform/profilesync/util/JsonFeignResponseUtilTest.java:29]: This is the  old class name and i dint added newly. i can create techdebt ticket to change the class names in the existing  repos

File: src/test/java/uk/gov/hmcts/reform/profilesync/domain/ProfileSyncAuditTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+import org.junit.Test;
+
+public class ProfileSyncAuditTest {
+
+    private String status = "success";
+
+    @Test
+    public void shouldPopulateFewFields() {
+        ProfileSyncAudit profileSyncAuditsync = new ProfileSyncAudit(LocalDateTime.now(), status);
```
[the remark this answers is not in the document]
Comment (KOTLAPRA23) [reply src/test/java/uk/gov/hmcts/reform/profilesync/domain/ProfileSyncAuditTest.java:17]: This test to verify only ProfileSyncAudit fileds and make sure nothing populated inside the ProfileSyncAuditDetails class.

**Pull request merged**